src/lib/datepicker/datepicker.ts

```diff
@@ -145,6 +145,9 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
   /** An input indicating the type of the custom header component for the calendar, if set. */
   @Input() calendarHeaderComponent: ComponentType<any>;
 
+  /** An input indicating the type of the custom footer component for the calendar, if set. */
+  @Input() calendarFooterComponent: ComponentType<any>;
```

Comment:
I don't know about this, I don't think we want to let users drop content in here. At least not the way it's currently set up in this PR, the content will just wind up floating below the calendar.

For now how about just putting a button in `mat-datepicker-content` and we can consider opening it up via an API at some point in the future

Reply:
Sure